lib/puppet/provider/cisco_interface_portchannel/nxapi.rb

```diff
@@ -0,0 +1,159 @@
+    :lacp_min_links,
+    :port_hash_distribution,
+  ]
+  INTF_PC_BOOL_PROPS = [
+    :lacp_graceful_convergence,
+    :lacp_suspend_individual,
+    :port_load_defer,
+  ]
+  INTF_PC_ALL_PROPS = INTF_PC_NON_BOOL_PROPS + INTF_PC_BOOL_PROPS
+
+  PuppetX::Cisco::AutoGen.mk_puppet_methods(:non_bool, self,
+                                            '@interface_portchannel',
+                                            INTF_PC_NON_BOOL_PROPS)
+  PuppetX::Cisco::AutoGen.mk_puppet_methods(:bool, self,
+                                            '@interface_portchannel',
+                                            INTF_PC_BOOL_PROPS)
+
+  def initialize(value={})
+    super(value)
+    @interface_portchannel =
+        Cisco::InterfacePortChannel.interfaces[@property_hash[:name]]
+    @property_flush = {}
+  end
+
+  def self.properties_get(interface_name, intf)
+    debug "Checking instance, #{interface_name}."
+    current_state = {
+      interface: interface_name,
+      name:      interface_name,
+      ensure:    :present,
+    }
+    # Call node_utils getter for each property
+    INTF_PC_NON_BOOL_PROPS.each do |prop|
+      current_state[prop] = intf.send(prop)
+    end
+    INTF_PC_BOOL_PROPS.each do |prop|
+      val = intf.send(prop)
+      if val.nil?
+        current_state[prop] = nil
+      else
+        current_state[prop] = val ? :true : :false
+      end
+    end
+    new(current_state)
+  end # self.properties_get
+
+  def self.instances
+    interfaces = []
+    Cisco::InterfacePortChannel.interfaces.each do |interface_name, intf|
+      interfaces << properties_get(interface_name, intf)
+    end
+    interfaces
+  end # self.instances
+
+  def self.prefetch(resources)
+    interfaces = instances
+    resources.keys.each do |name|
+      provider = interfaces.find { |intf| intf.instance_name == name }
+      resources[name].provider = provider unless provider.nil?
+    end
+  end # self.prefetch
+
+  def exists?
+    (@property_hash[:ensure] == :present)
+  end
+
+  def create
+    @property_flush[:ensure] = :present
+  end
+
+  def destroy
+    @property_flush[:ensure] = :absent
+  end
+
+  def instance_name
+    interface
+  end
+
+  def properties_set(new_interface=false)
+    INTF_PC_ALL_PROPS.each do |prop|
+      next unless @resource[prop]
+      send("#{prop}=", @resource[prop]) if new_interface
+      unless @property_flush[prop].nil?
+        @interface_portchannel.send("#{prop}=", @property_flush[prop]) if
+          @interface_portchannel.respond_to?("#{prop}=")
+      end
+    end
+  end
+
+  def flush
+    if @property_flush[:ensure] == :absent
+      @interface_portchannel.destroy
+      @interface_portchannel = nil
+    else
+      # Create/Update
+      if @interface_portchannel.nil?
+        new_interface = true
+        @interface_portchannel =
+            Cisco::InterfacePortChannel.new(@resource[:interface])
+      end
+      properties_set(new_interface)
+    end
+    puts_config
+  end
+
+  def puts_config
```

Comment:
No need to include this `puts_config` method.  We did this for some early providers but it's not needed.